packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -41,7 +52,54 @@ const getClassDeclaration = (
     name: declaration.name?.text ?? '',
     node: declaration,
     getHeritage: () => getHeritage(declaration, analyzer),
+    ...parseNodeJSDocInfo(declaration),
+    ...getClassMembers(declaration, analyzer),
+  });
+};
+
+/**
+ * Returns the `fields` and `methods` of a class.
+ */
+export const getClassMembers = (
+  declaration: ts.ClassDeclaration,
+  analyzer: AnalyzerInterface
+) => {
+  const fieldMap = new Map<string, ClassField>();
+  const methodMap = new Map<string, ClassMethod>();
+  declaration.members.forEach((node) => {
+    if (ts.isMethodDeclaration(node)) {
+      methodMap.set(
+        node.name.getText(),
+        new ClassMethod({
+          ...getMemberInfo(node),
+          ...getFunctionLikeInfo(node, analyzer),
+          ...parseNodeJSDocInfo(node),
+        })
+      );
+    } else if (ts.isPropertyDeclaration(node)) {
```

Comment:
> What about accessors? Are they handled in one of these two cases?

No, not handled yet. I added that to the punch list here: https://github.com/lit/lit/issues/2993

> And are function-valued properties handled as-is, or represented as methods?

Right now they're handed as function-valued fields, which feels right since they're not "methods" in the technical sense (often intended to be overridden at instance time where methods are not, etc).

However, for exported declarations, I am planning (unless you have thoughts otherwise) to handle `export const foo = () => {...}` as a `FunctionDeclaration` rather than a function-valued `VariableDeclaration`, since it is mostly unobservable/stylistic differences between that and `export function foo() {...}`.

Reply:
I think that's right. I was asking exactly because a field cannot be overridden with a method.

And for exported function variables as functions, yes also - I don't think the difference is observable at all.